package Alura.Exepciones;

public class TryWithResources {

    public static void main(String[] args) throws Exception {

        try (Conexion conexion = new Conexion()) {
            conexion.leerDatos();
        } catch (IllegalStateException ex) {
            System.out.println("Surgi贸 un error en la conexi贸n ");
        }
    }

}

class Conexion implements AutoCloseable {

    public Conexion() {
        System.out.println("Abriendo conexi贸n");
    }

    public void leerDatos() {
        System.out.println("Recibiendo datos");
        throw new IllegalStateException();
    }

    public void cerrar() {
        System.out.println("Cerrando conexi贸n ");
    }

    @Override
    public void close() throws Exception {
        System.out.println("Cerrando Conexion");
    }
}
